Assets/Scripts/Bosses/MechBossMissileMotion.cs

```diff
@@ -30,7 +31,7 @@ void Update()
     {
         if (!isFalling && Time.time - spawnTime >= riseTime)
         {
-            transform.position = new Vector3(playerLocationAtSpawn.x, fallHeight, transform.position.z);
+            transform.position = new Vector3(playerLocationAtSpawn.x + horizontalOffset, fallHeight, transform.position.z);
             Vector3 localScale = transform.localScale;
 		    localScale.y *= -1;
 		    transform.localScale = localScale;
```

Comment:
Since we're managing the `transform.position` directly, we'll want to make sure that the missile's RigidBody2D is set to `kinematic` if not already so

Reply:
What exactly does that mean?

Reply:
From the [docs](https://docs.unity3d.com/Manual/RigidbodiesOverview.html):

> In most cases, if a GameObject has a Rigidbody, you should use the Rigidbody properties to move the GameObject, instead of the Transform properties. The Rigidbody properties apply forces and torque from the physics system, which change the GameObject’s Transform; if you then change the Transform directly, Unity cannot correctly calculate the physics simulation, and you might see unwanted behavior. This is particularly true of rigid bodies with [Joints](https://docs.unity3d.com/Manual/joints-section.html).

> There are some cases where you might want the physics system to detect a GameObject, but not to control it. For example, you might want [Colliders](https://docs.unity3d.com/Manual/collision-section.html)
 to detect a GameObject, but you intend to control that GameObject’s movement and position via its [Transform](https://docs.unity3d.com/Manual/class-Transform.html).

> Movement in Unity that is not physics-based is called kinematic motion. The Rigidbody component has the property Is Kinematic which, when enabled, defines the attached GameObject as non-physics-based, and removes it from the control of the physics engine. This allows you to move it kinematically via the Transform without Unity’s physics simulation calculations overriding the changes.